### PR TITLE
feat/messages additions (in memory of Leonard Nimoy )

### PR DESCRIPTION
**Pull request by itschip**
[Tweak/messages additions](https://app.gitkraken.com/glo/card/fec49a0815d145279a5711df950cfbcc)

### CODE REVIEW

**Review by jfrader — APPROVED** [review]
Nice Job!

File: phone/src/apps/messages/components/modal/Conversation.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import useStyles from './modal.styles';
 import { MessageImageModal } from './MessageImageModal';
 import { useQueryParams } from '../../../../common/hooks/useQueryParams';
 import { MessageBubble } from './MessageBubble';
+import { act } from 'react-dom/test-utils';
```
Comment (TasoOneAsia) [inline phone/src/apps/messages/components/modal/Conversation.tsx:11]: Please remove

File: resources/server/sv_messages.ts [hunk 2]
```diff
@@ -384,6 +384,16 @@ async function createMessageGroupsFromPhoneNumbers(
   return { error: false, groupId };
 }
 
+// getting the participants from groupId.
+// this should return the source or and array of identifiers
+async function getIdentifiersFromParticipants(groupId: string) {
+  const query =
+    'SELECT participant_identifier FROM npwd_messages_groups WHERE group_id = ?';
+  const [results] = await pool.query(query, [groupId]);
+  console.log(<any[]>results);
```
Comment (TasoOneAsia) [inline resources/server/sv_messages.ts:393]: Remove console.log

**Review by TasoOneAsia — CHANGES_REQUESTED** [review]

**Comment by itschip** [comment]
```sql
unreadCount int(11) NOT NULL DEFAULT 0,
```

This is needed in your DB for `npwd_messages_group`, if you want to test.

File: phone/src/apps/messages/hooks/useMessageNotifications.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const { getMessageGroupById } = useMessages();
+
+  const setNotification = ({ groupId, message }) => {
+    const group = getMessageGroupById(groupId);
+    if (!group) return;
+
+    const id = `${NOTIFICATION_ID}:${groupId}`;
+
+    const notification = {
+      app: 'MESSAGS',
```
Comment (jfrader) [inline phone/src/apps/messages/hooks/useMessageNotifications.ts:28]:
```suggestion
      app: 'MESSAGES',
```

File: phone/src/os/notifications/providers/NotificationsProvider.tsx [hunk 4]
```diff
@@ -110,37 +106,29 @@ export function NotificationsProvider({ children }) {
...
-        updateNotification(exists, update);
-        return;
-      }
-      addNotification(add);
+  /**
+   * Checks if a notification exists by its `id`
+   */
+  const hasNotification = useCallback(
+    (id: string): INotification | null => {
+      console.log(id, notifications);
```
Comment (jfrader) [inline phone/src/os/notifications/providers/NotificationsProvider.tsx:121]:
```suggestion
```

**Review by jfrader — APPROVED** [review]

File: phone/src/common/typings/messages.ts [hunk 5]
```diff
@@ -26,3 +28,12 @@ export interface CreateMessageGroupResult {
   groupId?: string;
   mine?: boolean;
 }
+
+export interface CreateMessageBroadcast {
+  message: string;
+  groupName: string;
```
Comment (jfrader) [inline phone/src/common/typings/messages.ts:34]:
```suggestion
  groupName: string;
  groupId: string;
```

**Review by jfrader — APPROVED** [review]

File: phone/src/apps/messages/hooks/useMessageNotifications.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+  const setNotification = ({ groupId, message }) => {
+    const group = getMessageGroupById(groupId);
+    if (!group) return;
+
+    const id = `${NOTIFICATION_ID}:${groupId}`;
+
+    const notification = {
+      app: 'MESSAGES',
+      id,
```
Comment (jfrader) [inline phone/src/apps/messages/hooks/useMessageNotifications.ts:29]:
```suggestion
      id,
      sound: true,
```

**Review by jfrader — APPROVED** [review]

File: phone/src/apps/messages/components/modal/Conversation.tsx (hunk 1 above)
Comment (jfrader) [reply phone/src/apps/messages/components/modal/Conversation.tsx]: Done

File: resources/server/sv_messages.ts [hunk 7]
```diff
@@ -384,6 +402,24 @@ async function createMessageGroupsFromPhoneNumbers(
...
+  return groupId.split(':');
+}
+
+/**
+ * Sets the current message isRead to 0 for said player
+ * @param id
+ */
+async function setMessageRead(groupId: string, pSource: number) {
+  const query =
+    'UPDATE npwd_messages_groups SET unreadCount = 0 WHERE groupId = ? AND participant_identifier = ?';
```
Comment (jfrader) [inline resources/server/sv_messages.ts:417]:
```suggestion
    'UPDATE npwd_messages_groups SET unreadCount = 0 WHERE group_id = ? AND participant_identifier = ?';
```

**Pull request merged**